src/sage/graphs/generic_graph.py

```diff
@@ -15745,6 +15745,41 @@ def distance_all_pairs(self, by_weight=False, algorithm=None,
                                             algorithm=algorithm,
                                             weight_function=weight_function,
                                             check_weight=check_weight)[0]
+                                
+    def power(self, k):
+        r"""
+        Compute the kth power graph of an undirected, unweighted graph based on
+        shortest distances between nodes using BFS.
+
+        INPUT:
+        - graph: An undirected, unweighted graph.
+        - k: The maximum path length for considering edges in the power graph.
+
+        OUTPUT:
+        - The kth power graph based on shortest distances between nodes.
+
+        EXAMPLE:
+
+        sage: G = Graph([(0, 1), (1, 2), (2, 3), (3, 0), (2, 4), (4, 5)])
+        sage: k = 2
+        sage: PG = G.power(k)
+        sage: PG.edges()
```

Comment:
the reported order of edges may change. You can use `PG.edges(sort=True, labels=False)` here as vertex labels are comparable.